### PR TITLE
Double slash in URL makes git clone fail

### LINKED ISSUE #20462: Extra slash in URL makes git clone fail
### Description

Cloning a repository with a double slash fails:
```shell
$ git clone https://git.data.coop//halfd/new-website.git
Cloning into 'new-website'...
remote: Not found.
fatal: repository 'https://git.data.coop//halfd/new-website.git/' not found
```

### Gitea Version

1.16.8

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Using the docker image.

### Database

SQLite

### PR DESCRIPTION
**Pull request by stuzer05**
Allow urls prefixes with double slashes be valid

e.g. `http://localhost:3000//stuzer05/test_rights.git`

Closes #20462

### CODE REVIEW

File: routers/common/middleware.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package common
 
 import (
 	"fmt"
+	"github.com/go-chi/chi/v5"
```
Comment (lunny) [inline routers/common/middleware.go]: wrong import sequence

File: routers/common/middleware.go [hunk 2]
```diff
@@ -50,6 +51,28 @@ func Middlewares() []func(http.Handler) http.Handler {
...
+	handlers = append(handlers, func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+			var path string
+			rctx := chi.RouteContext(req.Context())
+			if rctx != nil && rctx.RoutePath != "" {
+				path = rctx.RoutePath
+			} else {
+				path = req.URL.Path
+			}
+			fmt.Println(strings.HasPrefix(path, "//"))
```
Comment (lunny) [inline routers/common/middleware.go]: Please remove trace log

File: routers/common/middleware.go [hunk 3]
```diff
@@ -50,6 +51,28 @@ func Middlewares() []func(http.Handler) http.Handler {
...
+			if len(path) > 1 && strings.HasPrefix(path, "//") {
```
Comment (lunny) [inline routers/common/middleware.go:63]: Suffix?

Comment (stuzer05) [reply routers/common/middleware.go:63]: Suffix is already being trimmed in `routers/common/middleware.go:52` (`middleware.StripSlashes`). Here is prefix trim (`//path/to`)

**Comment by wolfogre** [comment]
Hi @stuzer05, there's an oversight at work, we didn't connect PR #21333 to issue #21333, and this led to you overlooking the fact that there's already a PR addressing the same issue. Sorry for that.

That's not your fault, but I have to say this PR is duplicated and I have to close it. If you have different ideas, please feel free to reopen it.

**Comment by stuzer05** [comment]
> Hi @stuzer05, there's an oversight at work, we didn't connect PR #21333 to issue #20462, and this led to you overlooking the fact that there's already a PR addressing the same issue. Sorry for that.
> 
> That's not your fault, but I have to say this PR is duplicated and I have to close it. If you have different ideas, please feel free to reopen it.

My code looks cleaner for the same job, I think. It's mainly taken from `middleware.StripSlashes`. I can adjust it to strip all double slashes pretty easily

**Comment by wolfogre** [comment]
> > Hi @stuzer05, there's an oversight at work, we didn't connect PR #21333 to issue #20462, and this led to you overlooking the fact that there's already a PR addressing the same issue. Sorry for that.
> > That's not your fault, but I have to say this PR is duplicated and I have to close it. If you have different ideas, please feel free to reopen it.
> 
> My code looks cleaner for the same job, I think. It's mainly taken from `middleware.StripSlashes`. I can adjust it to strip all double slashes pretty easily

I understand, however, you know it's not a competition. We can also adjust the code in #21333. If you have better ideas, you are welcome to help review #21333.

**Comment by stuzer05** [comment]
> > > Hi @stuzer05, there's an oversight at work, we didn't connect PR #21333 to issue #20462, and this led to you overlooking the fact that there's already a PR addressing the same issue. Sorry for that.
> > > That's not your fault, but I have to say this PR is duplicated and I have to close it. If you have different ideas, please feel free to reopen it.
> > 
> > 
> > My code looks cleaner for the same job, I think. It's mainly taken from `middleware.StripSlashes`. I can adjust it to strip all double slashes pretty easily
> 
> I understand, however, you know it's not a competition. We can also adjust the code in #21333. If you have better ideas, you are welcome to help review #21333.

Sure! Don't get me wrong, I just want the code to be good, that's all)

**Pull request closed without merging**